packages/flutter_tools/test/general.shard/ios/devices_test.dart

```diff
@@ -75,6 +79,34 @@ void main() {
       });
     }
 
+    group('.dispose()', () {
+      testUsingContext(' kills all log readers & port forwarders', () async {
+        final IOSDevice device = IOSDevice('123');
+        final MockApplicationPackage applicationPackage1 = MockApplicationPackage();
+        final MockApplicationPackage applicationPackage2 = MockApplicationPackage();
+        when(applicationPackage1.name).thenReturn('flutterApp1');
+        when(applicationPackage2.name).thenReturn('flutterApp2');
+        final IOSDeviceLogReader logReader1 = IOSDeviceLogReader(device, applicationPackage1);
+        final IOSDeviceLogReader logReader2 = IOSDeviceLogReader(device, applicationPackage2);
+        final MockProcess mockProcess1 = MockProcess();
+        final MockProcess mockProcess2 = MockProcess();
+        logReader1.idevicesyslogProcess = mockProcess1;
+        logReader2.idevicesyslogProcess = mockProcess2;
+        final IOSDevicePortForwarder portForwarder = IOSDevicePortForwarder(device);
+        final MockForwardedPort mockForwardedPort = MockForwardedPort();
+        portForwarder.addForwardedPorts(<ForwardedPort>[mockForwardedPort]);
+        device.setLogReader(applicationPackage1, logReader1);
+        device.setLogReader(applicationPackage2, logReader2);
+        device.portForwarder = portForwarder;
+        device.dispose();
+        verify(mockProcess1.kill());
```

Comment:
Nit: there is a lot of code here, it would be easier to read if you separate the setup, the actions, and the expectations.

Reply:
done